packages/remix-react/browser.tsx

```diff
@@ -266,9 +273,10 @@ export function RemixBrowser(_props: RemixBrowserProps): ReactElement {
       mapRouteProperties,
     });
 
-    // As long as we didn't SSR a `HydrateFallback`, we can initialize immediately since
-    // there's no initial client-side data loading to perform
-    if (!didServerRenderFallback) {
+    // We can call initialize() immediately if the router doesn't have any
+    // loaders to run on hydration
+    if (router.state.initialized) {
+      routerInitialized = true;
```

Comment:
We don't need `didServerRenderFallback` anymore now that `router.state.initialized` is fixed up for partial hydration